File: backend/src/main/java/com/lawfirm/domain/matter/entity/Deadline.java
package com.lawfirm.domain.matter.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.lawfirm.common.base.BaseEntity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 期限提醒实体
 */
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@TableName("matter_deadline")
public class Deadline extends BaseEntity {

    /**
     * 项目ID
     */
    private Long matterId;

    /**
     * 期限类型：EVIDENCE_SUBMISSION-举证期, APPEAL-上诉期, REPLY-答辩期, EXECUTION_APPLICATION-执行申请期, HEARING-开庭期, OTHER-其他
     */
    private String deadlineType;

    /**
     * 期限名称
     */
    private String deadlineName;

    /**
     * 基准日期（如立案日期、开庭日期、判决日期等）
     */
    private LocalDate baseDate;

    /**
     * 期限日期
     */
    private LocalDate deadlineDate;

    /**
     * 提前提醒天数（默认7天）
     */
    private Integer reminderDays;

    /**
     * 是否已发送提醒
     */
    private Boolean reminderSent;

    /**
     * 提醒发送时间
     */
    private LocalDateTime reminderSentAt;

    /**
     * 状态：ACTIVE-有效, COMPLETED-已完成, EXPIRED-已过期, CANCELLED-已取消
     */
    private String status;

    /**
     * 完成时间
     */
    private LocalDateTime completedAt;

    /**
     * 完成人ID
     */
    private Long completedBy;

    /**
     * 期限说明
     */
    private String description;
}

File: frontend/apps/web-antd/src/api/matter/deadline.ts
// 使用 types.ts 中的 PageResult 定义
import type { PageResult } from './types';

import { requestClient } from '#/api/request';

// ==================== 类型定义 ====================

export interface DeadlineDTO {
  id: number;
  matterId: number;
  matterNo?: string;
  matterName?: string;
  deadlineType: string;
  deadlineTypeName?: string;
  deadlineName: string;
  baseDate?: string;
  deadlineDate: string;
  reminderDays?: number;
  reminderSent?: boolean;
  reminderSentAt?: string;
  status: string;
  statusName?: string;
  completedAt?: string;
  completedBy?: number;
  completedByName?: string;
  description?: string;
  daysRemaining?: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface DeadlineQueryDTO {
  pageNum?: number;
  pageSize?: number;
  matterId?: number;
  deadlineType?: string;
  status?: string;
  startDate?: string;
  endDate?: string;
}

export interface CreateDeadlineCommand {
  matterId: number;
  deadlineType: string;
  deadlineName: string;
  baseDate?: string;
  deadlineDate: string;
  reminderDays?: number;
  description?: string;
}

export interface UpdateDeadlineCommand {
  id: number;
  deadlineName?: string;
  deadlineDate?: string;
  reminderDays?: number;
  description?: string;
}

// 期限类型选项
export const DEADLINE_TYPE_OPTIONS = [
  { label: '举证期', value: 'EVIDENCE_SUBMISSION' },
  { label: '答辩期', value: 'DEFENSE' },
  { label: '上诉期', value: 'APPEAL' },
  { label: '申请执行期', value: 'EXECUTION_APPLICATION' },
  { label: '开庭日期', value: 'COURT_DATE' },
  { label: '仲裁开庭', value: 'ARBITRATION_DATE' },
  { label: '合同履行期', value: 'CONTRACT_PERFORMANCE' },
  { label: '其他', value: 'OTHER' },
];

// 状态选项
export const DEADLINE_STATUS_OPTIONS = [
  { label: '待处理', value: 'PENDING' },
  { label: '已完成', value: 'COMPLETED' },
  { label: '已过期', value: 'EXPIRED' },
];

// ==================== API ====================

/** 分页查询期限列表 */
export function getDeadlines(params: DeadlineQueryDTO) {
  return requestClient.get<PageResult<DeadlineDTO>>('/matter/deadlines/list', {
    params,
  });
}

/** 获取期限详情 */
export function getDeadlineById(id: number) {
  return requestClient.get<DeadlineDTO>(`/matter/deadlines/${id}`);
}

/** 根据项目ID查询期限列表 */
export function getDeadlinesByMatterId(matterId: number) {
  return requestClient.get<DeadlineDTO[]>(
    `/matter/deadlines/matter/${matterId}`,
  );
}

/** 创建期限提醒 */
export function createDeadline(data: CreateDeadlineCommand) {
  return requestClient.post<DeadlineDTO>('/matter/deadlines', data);
}

/** 自动创建期限提醒 */
export function autoCreateDeadlines(matterId: number) {
  return requestClient.post<void>(`/matter/deadlines/auto-create/${matterId}`);
}

/** 更新期限提醒 */
export function updateDeadline(data: UpdateDeadlineCommand) {
  return requestClient.put<DeadlineDTO>('/matter/deadlines', data);
}

/** 完成期限 */
export function completeDeadline(id: number) {
  return requestClient.post<DeadlineDTO>(`/matter/deadlines/${id}/complete`);
}

/** 删除期限提醒 */
export function deleteDeadline(id: number) {
  return requestClient.delete<void>(`/matter/deadlines/${id}`);
}
